eng/tools/doccheck/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/doc"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func filter(f fs.FileInfo) bool {
+	return !strings.HasSuffix(f.Name(), "_test.go")
+}
+
+func findAllSubDirectories(root string) []string {
+	ret := make([]string, 0)
+
+	filepath.Walk(root, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			panic(err)
+		}
+		if info.IsDir() && !strings.Contains(path, "internal") {
```

Comment:
You can apparently stop recursing in specific folders by returning SkipDir (just found out about this)

> The error result returned by the function controls how Walk continues. If the function returns the special value SkipDir, Walk skips the current directory

This would also prevent the unlikely problem that someone has *internal* in their string and it's not actually an `internal` folder. Also, gets rid of some unneeded calls.